models/mainModels.js

```diff
@@ -0,0 +1,72 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const bcrypt = require('bcrypt');
+
+// connect to atlas
+const mongoURI =
+  'mongodb+srv://tester:password5@cluster0.1sqlz.mongodb.net/BookshelfDB?retryWrites=true&w=majority';
+mongoose
+  .connect(mongoURI, {
+    // options for the connect method to parse the URI
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    // sets the name of the DB that our collections are part of
+    dbName: 'BookshelfDB',
+    useCreateIndex: true, //prevents deprecation warning
+  })
+  .then(() => console.log('Connected to Mongo DB.'))
+  .catch(() => console.log('Error in connection to Mongo DB'));
+
+// media schema
+const mediaSchema = new Schema({
+  title: String,
+  type: String,
+  currentStatus: String,
+});
+
+const SALT_WORK_FACTOR = 10;
+
+// user schema
+const userSchema = new Schema({
+  // User.profileInfo in controllers
+  profileInfo: {
+    username: {type: String, unique: true, required: true},
```

Comment:
profileInfo should also have schema for the following fields:

email
firstName
lastName

Reply:
reminder for rest of us: username will be fully lower-case